spec/middleware_spec.rb

```diff
@@ -352,22 +381,4 @@ def mock_app(options = {}, conditions = {}, custom_headers = {})
       end
     end
   end
-
-  it "does not get stuck rendering each request as pdf" do
```

Comment:
Per the commit message:

```
...The spec now fails, because all of the
middleware's instance variables are, from the outside, irrelevant: they
only matter for the `dup` instance, which is GC'd right away.
```

> While the thread safety test does assert on different types of requests/responses, I would argue that it should not be used to validate other behavior of the class, as thread safety testing is in and of itself complicated enough to warrant not using the same test for implicitly asserting other behavior.

These tests weren't testing "other" behavior, they were testing (or trying to test) that PDFkit properly tracks whether a given request is in PDF mode - the same problem that initiated this PR. I think these tests were an attempt to protect against threading issues, but they never actually reproduced the problem, because they were single-threaded.